### PR TITLE
guard ncp spinel tests with OPENTHREAD_BUILD_TESTS

### PR DESCRIPTION
**Pull request by mszczodrak**

### CODE REVIEW

File: src/ncp/ncp_base.cpp [hunk 1]
```diff
@@ -568,7 +568,7 @@ void NcpBase::HandleCommandPropertyGet(uint8_t header, spinel_prop_key_t key)
 void NcpBase::HandleCommandPropertySet(uint8_t header, spinel_prop_key_t key, const uint8_t *value_ptr,
                                        uint16_t value_len)
 {
-    int i;
+    unsigned i;
```
Comment (darconeous) [inline src/ncp/ncp_base.cpp]: Why this change?

Comment (mszczodrak) [inline src/ncp/ncp_base.cpp]: to remove compilation warning between comparing signed with unsigned

**Comment by darconeous** [comment]
👍 LGTM

**Comment by jwhui** [comment]
LGTM :+1:
